com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/linking/ImportedNamesTypesAdapter.java

```diff
@@ -31,7 +31,7 @@
  */
 public class ImportedNamesTypesAdapter extends ImportedNamesAdapter {
 
-  private final Map<QualifiedName, Set<EClass>> importedNamesTypes = Maps.newHashMap();
+  private final Map<QualifiedName, Set<EClass>> importedNamesTypes = CacheManager.getInstance().createMapCache("ImportedNamesTypesAdapter#importedNamesTypes"); //$NON-NLS-1$

```

Comment:
this isn't really a cache either. We collect the names of objects imported in the given resource and then write it to the db...